dispatcher.go

```diff
@@ -230,3 +261,30 @@ func (d dispatcher) Stop() error {
 
 	return nil
 }
+
+// getUniqueOutbounds helps ensure we only start/stop the same outbound once
+func (d dispatcher) getUniqueOutbounds() []transport.Outbound {
+	seen := make(map[transport.Outbound]struct{})
+
+	addIfUnique := func(outbound transport.Outbound) {
+		if outbound == nil {
+			return
+		}
+
+		if _, ok := seen[outbound]; !ok {
+			seen[outbound] = struct{}{}
+		}
+	}
+
+	for _, outs := range d.outbounds {
+		addIfUnique(outs.Unary)
+		addIfUnique(outs.Oneway)
+	}
+
+	unique := make([]transport.Outbound, 0, len(seen))
+	for outbound := range seen {
```

Comment:
Instead of doing this in two stages, just append to `unique` inside
`addIfUnique`.